src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs

```diff
@@ -31,31 +30,31 @@ private enum WellKnownAttributeFlags
             IsCallerFilePath = 0x1 << 5,
             IsCallerLineNumber = 0x1 << 6,
             IsCallerMemberName = 0x1 << 7,
-            NotNullWhenTrue = 0x1 << 8,
-            NotNullWhenFalse = 0x1 << 9,
-            AssertsTrue = 0x1 << 10,
-            AssertsFalse = 0x1 << 11,
         }
 
         private struct PackedFlags
         {
             // Layout:
-            // |.....|n|rr|cccccccccccc|vvvvvvvvvvvv|
+            // |...|fffffffff|n|rr|cccccccc|vvvvvvvv|
             // 
-            // v = decoded well known attribute values. 12 bits.
-            // c = completion states for well known attributes. 1 if given attribute has been decoded, 0 otherwise. 12 bits.
+            // v = decoded well known attribute values. 8 bits.
+            // c = completion states for well known attributes. 1 if given attribute has been decoded, 0 otherwise. 8 bits.
             // r = RefKind. 2 bits.
             // n = hasNameInMetadata. 1 bit.
+            // f = FlowAnalysisAnnotations. 9 bits (8 value bits + 1 completion bit).
 
             private const int WellKnownAttributeDataOffset = 0;
-            private const int WellKnownAttributeCompletionFlagOffset = 12;
-            private const int RefKindOffset = 24;
+            private const int WellKnownAttributeCompletionFlagOffset = 8;
+            private const int RefKindOffset = 16;
+            private const int FlowAnalysisAnnotationsOffset = 20;
 
             private const int RefKindMask = 0x3;
-            private const int WellKnownAttributeDataMask = 0xFFF;
+            private const int WellKnownAttributeDataMask = 0xFF;
```

Comment:
Is this going to break with code that was produced with the preview's we shipped already?

Reply:
The bit fields are simply populated and used at compile-time and are not persisted.

---
In reply to: [286166940](https://github.com/dotnet/roslyn/pull/35846#discussion_r286166940) [](ancestors = 286166940)